src/utils/ddb/ddb_vos.c

```diff
@@ -1771,29 +1771,148 @@ sync_cb(struct ddbs_sync_info *info, void *cb_args)
 		 */
 		blob_size = info->dsi_cluster_nr * info->dsi_cluster_size;
 	} else {
-		blob_size = pool_info->spi_blob_sz[st];
-		smd_pool_free_info(pool_info);
+		blob_size = pinfo->spi_blob_sz[st];
+		if (pool_info == NULL)
+			smd_pool_free_info(pinfo);
 	}
 
 	/* Try to delete the target first */
-	rc = smd_pool_del_tgt(pool_id, info->dsi_hdr->bbh_vos_id, st);
+	rc = smd_pool_del_tgt(pool_id, tgt_id, st);
 	if (!SUCCESS(rc))
 		/* Ignore error for now ... might not exist*/
 		D_WARN("delete target failed: " DF_RC "\n", DP_RC(rc));
 
-	rc = smd_pool_add_tgt(pool_id, info->dsi_hdr->bbh_vos_id, info->dsi_hdr->bbh_blob_id, st,
+	rc = smd_pool_add_tgt(pool_id, tgt_id, blob_id, st,
 			      blob_size, 0, false);
 	if (!SUCCESS(rc)) {
 		D_ERROR("add target failed: "DF_RC"\n", DP_RC(rc));
 		args->sync_rc = rc;
 		return;
 	}
 
+	if (st == SMD_DEV_TYPE_META) {
+		rc = smd_pool_mark_ready(pool_id);
```

Comment:
This should be avoided by setting "recreate" flag as "true" in above smd_pool_add_tgt().